test/modules/obsagent-client/src/main.rs

```diff
@@ -120,91 +43,27 @@ fn init_logging() {
-    let ud_counter = meter
-        .i64_up_down_counter("i64_up_down_counter_example")
-        .init();
-    let value_recorder = meter
-        .f64_value_recorder("f64_value_recorder_example")
-        .init();
-
-    // Init asynchronous instruments
-    let sum = Arc::new(Mutex::new(0));
-    let sum_clone = Arc::clone(&sum);
-    let sum_observer_cb = move |res: ObserverResult<u64>| {
-        if let Ok(sum_clone) = sum_clone.try_lock() {
-            res.observe(*sum_clone, &[]);
-        } else {
-            error!("try_lock failed")
-        }
-    };
-    let _sum_observer = meter
-        .u64_sum_observer("u64_sum_observer_example", sum_observer_cb)
-        .init();
-    let ud_sum = Arc::new(Mutex::new(0));
-    let ud_sum_clone = Arc::clone(&ud_sum);
-    let ud_sum_observer_cb = move |res: ObserverResult<i64>| {
-        if let Ok(ud_sum_clone) = ud_sum_clone.try_lock() {
-            res.observe(*ud_sum_clone, &[]);
-        } else {
-            error!("try_lock failed")
-        }
-    };
-    let _ud_sum_observer = meter
-        .i64_up_down_sum_observer("i64_up_down_sum_observer_example", ud_sum_observer_cb)
-        .init();
-    let value: Arc<Mutex<f64>> = Arc::new(Mutex::new(0.0));
-    let value_clone = Arc::clone(&value);
-    let value_observer_cb = move |res: ObserverResult<f64>| {
-        if let Ok(value_clone) = value_clone.try_lock() {
-            res.observe(*value_clone, &[]);
-        } else {
-            error!("try_lock failed")
-        }
-    };
-    let _value_observer = meter
-        .f64_value_observer("f64_value_observer_example", value_observer_cb)
-        .init();
+    #[cfg(feature = "otel")]
+    let otel_fut = otel_client::run(config.clone());
+    #[cfg(feature = "prom")]
+    let prom_fut = prometheus_server::run(config.clone());
 
-    // Loop, updating each metric value at the configured update rate
-    let mut rng = rand::thread_rng();
-    let term = Arc::new(AtomicBool::new(false));
-    signal_hook::flag::register(signal_hook::consts::SIGTERM, Arc::clone(&term))?;
-    while !term.load(Ordering::Relaxed) {
-        counter.add(1, &[]);
-        ud_counter.add(1, &[]);
-        value_recorder.record(rng.gen::<f64>(), &[]);
-        {
-            if let Ok(mut sum) = sum.try_lock() {
-                *sum += 1;
-            } else {
-                error!("try_lock failed");
-            }
-        }
-        {
-            if let Ok(mut ud_sum) = ud_sum.try_lock() {
-                *ud_sum += 1;
-            } else {
-                error!("try_lock failed");
-            }
-        }
-        {
-            if let Ok(mut value) = value.try_lock() {
-                *value = rng.gen::<f64>();
-            } else {
-                error!("try_lock failed");
-            }
+    cfg_if::cfg_if! {
+        if #[cfg(all(feature="otel", feature = "prom"))] {
+            let (otel_result, prom_result) = futures::join!(otel_fut, prom_fut);
```

Comment:
It might be better to run these benchmarking components on their own threads so that one future doesn't get greedy. Right now they are sharing one thread. Not sure if this will adversely affect the benchmarking.

Reply:
Ah I see you are spawning inside the futures. The pattern I am familiar with is to spawn on the run() command.

Reply:
Updated main.rs to spawn on `otel_client::run()` and `prometheus_server::run()`.